src/managers/TimestampManager.cpp

```diff
@@ -172,4 +174,155 @@ TimestampManager::getTimestampsByGlobalId(int globID, int timestampA,
+                appendBits(builder, val, 5);
+            } else if(val >= -63 && val <= 64){
+                appendBits(builder, 0b1110, 4);
+                appendBits(builder, val, 7);
+            } else if(val >= -255 && val <= 256){
+                appendBits(builder, 0b11110, 5);
+                appendBits(builder, val, 9);
+            } else if(val >= -1023 && val <= 1024){
+                appendBits(builder, 0b111110, 6);
+                appendBits(builder, val, 11);
+            } else if(val >=  -4095 && val <= 4096){
+                appendBits(builder, 0b1111110, 7);
+                appendBits(builder, val, 13);
+            } else {
+                appendBits(builder, 0b1111111, 7);
+                appendBits(builder, val, 32);
+            }
+        }
+    );
+
+    // ===== SCHEMA 2 =====:
+    compressionSchemes.emplace_back([](BitVecBuilder* builder, int val) {
+            if(val >= -63 && val <= 64){
+                appendBits(builder, 0b10, 2);
+                appendBits(builder, val, 7);
+            } else if(val >= -255 && val <= 256){
+                appendBits(builder, 0b110, 3);
+                appendBits(builder, val, 9);
+            } else if(val >= -2047 && val <= 2048){
+                appendBits(builder, 0b1110, 4);
+                appendBits(builder, val, 12);
+            } else {
+                appendBits(builder, 0b1111, 4);
+                appendBits(builder, val, 32);
+            }
+        }
+    );
+
+
+    // =====================================================
+    // =============== ADD MORE SCHEMAS HERE ===============
+    // =====================================================
+}
+
+std::vector<uint8_t> TimestampManager::binaryCompressGlobOffsets
+        (const std::vector<std::pair<int, int>>& offsets) {
+    int size = 0;
+    bool firstScheme = true;
+    std::vector<unsigned char> bestCompression;
+    for (int schemeID = 0; schemeID < compressionSchemes.size(); schemeID++) {
+        BitVecBuilder builder;
+        builder.currentByte = 0;
+        builder.remainingBits = 8;
+        builder.bytesCounter = 0;
+        const auto scheme = compressionSchemes.at(schemeID);
+
+        // Compress scheme ID.
+        // Always spend 8 bits on this value, as, when decompressing, we
+        // don't have a compression scheme for the first value.
+        // This allows for up to 256 different schemas.
+        appendBits(&builder, schemeID, 8);
+
+        for (const auto &elem: offsets) {
+            std::vector<int> temp;
+            temp.push_back(elem.first);
+            temp.push_back(elem.second);
+            for (const auto &t: temp) {
+                scheme(&builder, t);
+            }
+        }
+
+        // Update the chosen compression if it's better
+        size = builder.bytes.size();
+        if (size < bestCompression.size() || firstScheme) {
+            bestCompression = std::move(builder.bytes);
+            firstScheme = false;

```

Comment:
Check for bestcompression empty, instead of first scheme?

Reply:
done